lib/IRGen/GenCall.cpp

```diff
@@ -338,10 +338,12 @@ llvm::CallingConv::ID irgen::expandCallingConv(IRGenModule &IGM,
 
 static void addIndirectResultAttributes(IRGenModule &IGM,
                                         llvm::AttributeList &attrs,
-                                        unsigned paramIndex, bool allowSRet) {
+                                        unsigned paramIndex, bool allowSRet,
+                                        bool noCapture = true) {
```

Comment:
Also, this adds the `nocapture` attribute in _all the places_ `addIndirectResultAttributes` is used, which I'm not sure is always correct. I'm surprised more tests didn't fail because of this.

Reply:
Yeah, I'm not quite sure.  Swift doesn't currently expose the ability to rely on an escaped pointer into a value, at least in user code — it can happen in hidden ways, though, like in weak references.